miden/src/cli/data.rs

```diff
@@ -72,18 +93,134 @@ impl InputFile {
+            .map(|v| {
+                v.parse::<u64>()
+                    .map_err(|e| format!("failed to parse advice stack value `{v}` - {e}"))
+            })
+            .collect::<Result<Vec<_>, _>>()
+    }
+
+    /// Parse advice map data from the input file.
+    fn parse_advice_map(&self) -> Result<Option<HashMap<[u8; 32], Vec<Felt>>>, String> {
+        match &self.advice_map {
+            Some(advice_map) => {
+                let map = advice_map
+                    .iter()
+                    .map(|(k, v)| {
+                        // decode hex key
+                        let mut key = [0u8; 32];
+                        hex::decode_to_slice(k, &mut key)
+                            .map_err(|e| format!("failed to decode advice map key `{k}` - {e}"))?;
+
+                        // convert values to Felt
+                        let values = v
+                            .iter()
+                            .map(|v| {
+                                Felt::try_from(*v).map_err(|e| {
+                                    format!(
+                                        "failed to convert advice map value `{v}` to Felt - {e}"
+                                    )
+                                })
+                            })
+                            .collect::<Result<Vec<_>, _>>()?;
+                        Ok((key, values))
+                    })
+                    .collect::<Result<HashMap<[u8; 32], Vec<Felt>>, String>>()?;
+                Ok(Some(map))
+            }
+            None => Ok(None),
+        }
+    }
+
+    /// Parse merkle store data from the input file.
+    fn parse_merkle_store(&self) -> Result<Option<MerkleStore>, String> {
+        match &self.merkle_store {
+            Some(merkle_data) => {
+                let mut merkle_store = MerkleStore::default();
+                for data in merkle_data {
+                    match data {
+                        MerkleData::MerkleTree(data) => {
+                            let leaves = Self::parse_merkle_tree(data)?;
+                            merkle_store = merkle_store.with_merkle_tree(leaves).map_err(|e| {
+                                format!("failed to add merkle tree to merkle store - {e}")
+                            })?;
+                        }
+                        MerkleData::SparseMerkleTree(data) => {
+                            let entries = Self::parse_sparse_merkle_tree(data)?;
+                            merkle_store =
+                                merkle_store.with_sparse_merkle_tree(entries).map_err(|e| {
+                                    format!(
+                                        "failed to add sparse merkle tree to merkle store - {e}"
+                                    )
+                                })?;
+                        }
+                    }
+                }
+                Ok(Some(merkle_store))
+            }
+            None => Ok(None),
+        }
+    }
```

Comment:
It becomes a bit harder to read if we have too many indentation levels. Consider resolving the match patterns in the same indentation. Example:

```rust
fn parse_merkle_store(&self) -> Result<Option<MerkleStore>, String> {
    let merkle_data = match &self.merkle_store {
        Some(d) => d,
        None => return Ok(None),
    };
    (...)
```

Reply:
good catch. I've updated this and also `parse_advice_map` which had excessive indentation / nesting.